### PR TITLE
sso improvements

### PR DESCRIPTION
**Pull request by SamRemis**
*Issue #, if available:*
#2591 
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/Credentials/CredentialProviderTest.php [hunk 1]
```diff
@@ -1076,6 +1077,87 @@ public function testSsoProfileProvider()
         }
     }
 
+    public function testSsoProfileProviderWithNewFileFormat()
+    {
+        $dir = $this->clearEnv();
+        $expiration = time() + 1000;
+        $ini = <<<EOT
+[profile default]
```
Comment (stobrien89) [inline tests/Credentials/CredentialProviderTest.php:1085]: Should be `[default]`

**Review by stobrien89 — COMMENTED** [review]
Changes make sense to me. just had a couple of comments

File: src/Credentials/CredentialProvider.php [hunk 2]
```diff
@@ -321,78 +321,100 @@ public static function instanceProfile(array $config = [])
      *
      * @return callable
      */
-    public static function sso($ssoProfileName, $filename = null, $config = [])
-    {
+    public static function sso($ssoProfileName = 'profile default',
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Credentials/CredentialProvider.php:324]: We do now; I can change this

File: src/Credentials/CredentialProvider.php [hunk 3]
```diff
@@ -321,78 +321,100 @@ public static function instanceProfile(array $config = [])
...
         return function () use ($ssoProfileName, $filename, $config) {
             if (!@is_readable($filename)) {
                 return self::reject("Cannot read credentials from $filename");
             }
             $profiles = self::loadProfiles($filename);
-            if (!isset($profiles[$ssoProfileName])) {
+
+            if (isset($profiles[$ssoProfileName])) {
+                $ssoProfile = $profiles[$ssoProfileName];
+            } elseif (isset($profiles['profile ' . $ssoProfileName])) {
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Credentials/CredentialProvider.php:338]: yes :) I can add that to the PR

File: src/Credentials/CredentialProvider.php [hunk 4]
```diff
@@ -321,78 +321,100 @@ public static function instanceProfile(array $config = [])
...
+                $ssoProfileName = 'profile ' . $ssoProfileName;
+                $ssoProfile = $profiles[$ssoProfileName];
+            } else {
                 return self::reject("Profile {$ssoProfileName} does not exist in {$filename}.");
             }
-            $ssoProfile = $profiles[$ssoProfileName];
+
+            $ssoCredentials = null;
```
[the remark this answers is not in the document]
Comment (SamRemis) [reply src/Credentials/CredentialProvider.php:345]: Yes, for scope reasons.  This is used by some shared logic and set in either the if statement or the else statement below.  The alternative is duplicating code for the return statement, which is a potential option if we don't want to set them, especially if we expect the paths to diverge more in the future (I don't, but who knows)

File: tests/Credentials/CredentialProviderTest.php (hunk 1 above)
Comment (SamRemis) [reply tests/Credentials/CredentialProviderTest.php:1085]: I left this here to test that 'profile ' gets stripped and this gets used without having to add a basically duplicated test; should I get rid of it?

File: src/Credentials/CredentialProvider.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (stobrien89) [reply src/Credentials/CredentialProvider.php:338]: Niiice

**Review by stobrien89 — APPROVED** [review]

**Pull request merged**